sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzePartitionCommand.scala

```diff
@@ -140,7 +140,13 @@ case class AnalyzePartitionCommand(
     val df = tableDf.filter(Column(filter)).groupBy(partitionColumns: _*).count()
 
     df.collect().map { r =>
-      val partitionColumnValues = partitionColumns.indices.map(r.get(_).toString)
+      val partitionColumnValues = partitionColumns.indices.map { i =>
+        if (r.isNullAt(i)) {
+          ExternalCatalogUtils.DEFAULT_PARTITION_NAME
```

Comment:
do we need to chang the read path? i.e. where we use these statistics.

Reply:
I don't think so, as the same situation would happen if Hive's statistics are used instead of the ones computed by Spark